src/compass/s1_cslc_qa.py

```diff
@@ -659,4 +659,5 @@ def _get_land_mask(epsg_cslc: int, geotransform: tuple, shape_mask: tuple):
                       transform=tform_bbox,
                       out_shape=shape_mask,
                       dtype=np.uint8, default_value=1)
+    
```

Comment:
I can't comment directly on the code above for this PR since it wasn't changed:


```python

    # Make sure the CRS of the bounding box is the same as the GeoDataFrame
    bbox = gpd.GeoSeries([bbox], crs=f'EPSG:{epsg_cslc}')
    if epsg_cslc != 3413:
        bbox_3413 = bbox.to_crs('EPSG:3413')
```
this creates an undefined variable when `epsg_cslc == 3413`


and less, critical, but making a `GeoSeries` with one shapely polygon seems unnecessary since we're just re-selecting the polygon to do `.intersection`

Reply:
good catch. I've revised the implementation based on `ogr`, `osr`, and `gdal`.